### ISSUE TITLE
VariableForce : undefined method `[]' for #<Enumerator: "params":chars>

### ISSUE THREAD

**Issue by mullermp** [bbatsov/rubocop#3614]
While trying to access "params" or the "request" hash in a Rails controller method, Rubocop fails with:

```
An error occurred while VariableForce cop was inspecting /my/project/path/my_controller.rb.
undefined method `[]' for #<Enumerator: "params":chars>
```

Commenting out some code (shown below), I can get Rubocop to pass:

    requests = params[:fraud_information].split(/\r?\n/)
    notes = params[:notes]

--------

## Expected behavior

Rubocop should pass. It is legal to access a hash with a symbol.

## Actual behavior

Rubocop fails, as shown above, and prints an error while inspecting.

## Steps to reproduce the problem

Create a controller that a view can POST to, and then try to access the parameters with a symbol.

## RuboCop version

0.42.0 (using Parser 2.3.1.2, running on ruby 1.9.3 x86_64-linux)

**Comment by mullermp** [comment]
Just as a note, my company's sources do not have access to the newest version of Rubocop (we have a build management system that contains safe gems). I did dig through the changelogs and didn't notice anything that may have fixed this.

**Comment by savef** [comment]
`VariableForce` isn't a cop so that error message will need looking into.

The code you posted doesn't cause any errors for me, on `master` or `v0.42.0`. Could you post a more complete version and a full path that we can use to replicate the issue? Perhaps a config too (or preferably just see if you can replicate the issue with the default config).

>Any workarounds for now would be appreciated, even if there's a way to skip this cop for this controller, or skip this particular action.

These might not work since it's not a cop that seems to be failing, but to skip a cop for a controller you can add `Exclude: [app/controllers/examples_controller.rb]` to your config for that cop. Or to just skip a particular action you can append a comment to the method, such as `def update # rubocop:disable Metrics/CyclomaticComplexity`. Try adding the `--debug` switch to get a stack trace, that might help you pin down a cop to try these techniques with.

**Comment by mullermp** [comment]
Here's a paste of my config https://dpaste.de/U249
Controller: https://dpaste.de/oDOq
Stack trace (ignore the random failure at the end) https://dpaste.de/y3iS

appending ( # rubocop:disable ) next to my method definition still gives me the same failure.

I'm not sure which Cop to exclude from. Excluding from All cops seems to work though, but this is a bad idea.

**Comment by savef** [comment]
Ah! I just noticed you're on Ruby 1.9, support for that was dropped in `v0.42.0`, see [the CHANGELOG](https://github.com/bbatsov/rubocop/blob/a644414536a7715e1aecf247edf9fdad963e2ee6/CHANGELOG.md#changes-2).

**Comment by mullermp** [comment]
Unfortunately I'm stuck on Ruby 1.9 because I use JRuby and it's configured to use it as such. 

Is there a particular Cop that is causing this issue, so that I can exclude this file from it? I would rather not exclude from AllCops

**Comment by savef** [comment]
You might be better off sticking with `v0.41.2`, you'll probably just notice more similar crashes going forward.

Judging from the source of the exception (`VariableForce`) the issue is in a helper class so there might not be a single cop you can exclude. `Lint/UselessAssignment` is mentioned in the stack trace though, so try that one.

Before I spotted your Ruby version I had run my RuboCop on your code and `Lint/UselessAssignment` was the only offence. So if excluding that cop does fix your exception it probably means the error isn't with the cop's checking code but with the error reporting code. This may mean every time you have a _useless assignment_ in any file you will see these crashes, try adding one to see.

I know nothing of JRuby but can't you switch to JRuby 9k for more modern MRI syntax support?

**Comment by mullermp** [comment]
Looks like UselessAssignment is the root cause. Thank you for that. I commented out some useless variables (that I intended to use later but aren't using now) and the build appears to pass. So I suppose when this happens, I will know how to avoid it. I may also end up downgrading, or just adding that exception to Lint/UselessAssignment.

I can't switch the Ruby version. My company's build system is very complex and packages depend on other packages; it becomes a real mess come migration time.

**Comment by savef** [comment]
Good luck! I suppose this issue can be closed now. :smile_cat:

**Comment by bbatsov** [comment]
@mullermp I'd suggest downgrading as well.
